serverlessworkflow/modules/ROOT/pages/eventing/event-correlation-with-workflows.adoc

```diff
@@ -1 +1,201 @@
+        {
+          "contextAttributeName": "userid"
+        }
+      ]
+    },
+    {
+      "name": "validateAccountEmailEvent",
+      "source": "workflow",
+      "type": "validateAccountEmail"
+    },
+    {
+      "name": "validatedAccountEmailEvent",
+      "source": "workflow",
+      "type": "validatedAccountEmail",
+      "correlation": [
+        {
+          "contextAttributeName": "userid"
+        }
+      ]
+    },
+    {
+      "name": "activateAccountEvent",
+      "source": "workflow",
+      "type": "activateAccount"
+    },
+    {
+      "name": "activatedAccountEvent",
+      "source": "workflow",
+      "type": "activatedAccount",
+      "correlation": [
+        {
+          "contextAttributeName": "userid"
+        }
+      ]
+    }
+  ]
+----
+
+A workflow can be created by consuming events as defined by the Event state <<ref-new-user-account-state,`New User Account Request`>> which has a reference to the event <<ref-correlation-event-definition,`newAccountEvent`>> containing a correlation definition for attribute `userid`.
+
+[[ref-new-user-account-state]]
+.`New User Account Request` state definition
+[source,json]
+----
+{
+  "name": "New User Account Request",
+  "type": "event",
+  "onEvents": [
+    {
+      "eventRefs": [
+        "newAccountEvent"
+      ]
+    }
+  ],
+  "transition": "Validate User Email"
+}
+----
+
+When a new event of type <<ref-new-user-account-event,`newAccountEventType`>> is consumed the workflow is created and then any other events consumed by this workflow should contain the same correlation attribute `userid` with the same value `12345` set, which is used to evaluate and match the workflow instance to continue the execution.
+
+[[ref-new-user-account-event]]
+.New Incoming Start Event `newAccountEvent`
+[source,json]
+----
+{
+  "specversion": "0.3",
+  "id": "1d174d25-46ac-4785-bc76-457c2d37d2fe",
+  "source": "",
+  "type": "newAccountEventType",
+  "time": "2022-07-25T16:30:35.461988261-03:00",
+  "userid": "12345",
+  "data": {
+    "email": "test@test.com",
+    "userId": "12345"
+  }
+}
+----
+
+[NOTE]
+====
+It is not supported to correlate multiple events together, events are evaluated with correlations independently.
+====
+
+=== Producing and consuming events with correlation
+
+Following the link:{kogito_sw_examples_url}/serverless-workflow-correlation-quarkus/src/main/resources/correlation.sw.json[workflow definition], it is possible to define other events that are supposed to be published and consumed by this workflow, given example uses Callback States, such as the  <<ref-validade-user-email-state,`Validate User Email`>>, indicating once the workflow execution achieves this state, it publishes an event of type <<ref-correlation-event-definition,`validateAccountEmailEvent`>> and wait to receive an event of type <<ref-correlation-event-definition,`validatedAccountEmailEvent`>>, more information on xref:eventing/working-with-callbacks.adoc[Working with callbacks].
```

Comment:
I guess Heena will say that, but I think you can use more "." rather than "," in this parragraph ;)

Reply:
:+1: